src/model/delta/insertdelta.js

```diff
@@ -95,7 +104,44 @@ register( 'insert', function( position, nodes ) {
 	delta.addOperation( insert );
 	this.document.applyOperation( insert );
 
+	// When element is a DocumentFragment we need to move its markers to Document#markers.
+	if ( nodes instanceof DocumentFragment ) {
+		for ( const marker of nodes.markers ) {
+			const doc = this.document;
+			const range = moveRange( marker[ 1 ], position );
+			const markerDelta = new MarkerDelta();
+			const markerOperation = new MarkerOperation( marker[ 0 ], null, range, doc.markers, doc.version );
+
+			this.addDelta( markerDelta );
+			markerDelta.addOperation( markerOperation );
+			doc.applyOperation( markerOperation );
+		}
+	}
+
 	return this;
 } );
 
 DeltaFactory.register( InsertDelta );
+
+// Moves range relative to given position.
+//
+// @param {module:engine/model/range~Range} range
+// @param {module:engine/model/position~Position} position
+// @returns {module:engine/model/range~Range} Moved range.
+function moveRange( range, position ) {
```

Comment:
If you use `_getTransformedByMove`, `sourcePosition` should be the first position in document fragment ( `Position.createAt( range.root )` ), `targetPosition` should be equal to passed `position` and `howMany` should be number of document fragment children.

Or you can go with `_getCombined`:
```javascript
return new Range(
	range.start._getCombined( Position.createAt( range.root ), position ),
	range.end._getCombined( Position.createAt( range.root ), position )
);
```

Try those and let me know if those worked.